build.gradle.kts

```diff
@@ -0,0 +1,253 @@
+
+    compileOnly("micdoodle8.mods:galacticraft-api:$gcVersion")
+    compileOnly("micdoodle8.mods:galacticraft-core:$gcVersion")
+    compileOnly("micdoodle8.mods:galacticraft-planets:$gcVersion")
+    compileOnly("micdoodle8.mods:micdoodlecore:$gcVersion")
+
+    compileOnly(fg.deobf("mezz.jei:jei_${mcVersion}:${jeiVersion}:api"))
+    runtimeOnly(fg.deobf("mezz.jei:jei_${mcVersion}:${jeiVersion}"))
+}
+
+tasks.processResources {
+    //includeEmptyDirs = false
+    inputs.properties(
+        "advRocketryVersion" to project.version,
+        "mcVersion" to mcVersion,
+        "libVulpesVersion" to libVulpesVersion
+    )
+
+    filesMatching("mcmod.info") {
+        expand(
+            "advRocketryVersion" to project.version,
+            "mcVersion" to mcVersion,
+            "libVulpesVersion" to libVulpesVersion
+        )
+    }
+
+    exclude("**/*.sh")
+}
+
+tasks.register("cloneLibVulpes") {
+    group = "build setup"
+    doLast {
+        val libVulpesRepo: String by project
+        val libVulpesBranch: String? by project
+
+        val repo = Grgit.clone {
+            dir = "$projectDir/libVulpes"
+            uri = libVulpesRepo
+            if(libVulpesBranch != null)
+                refToCheckout = libVulpesBranch
+        }
+        println("Cloned libVulpes repository from $libVulpesRepo (current branch: ${repo.branch.current().name})")
+    }
+}
+
+val currentJvm: String = Jvm.current().toString()
+println("Current Java version: $currentJvm")
+
+val gitHash: String by lazy {
+    val hash: String = if (File(projectDir, ".git").exists()) {
+        val repo = Grgit.open(mapOf("currentDir" to project.rootDir))
+        repo.log().first().abbreviatedId
+    } else {
+        "unknown"
+    }
+    println("GitHash: $hash")
+    return@lazy hash
+}
+
+// Name pattern: [archiveBaseName]-[archiveAppendix]-[archiveVersion]-[archiveClassifier].[archiveExtension]
+tasks.withType(Jar::class) {
+    archiveAppendix.set(mcVersion)
+    manifest {
+        attributes(
+                "Built-By" to System.getProperty("user.name"),
+                "Created-By" to currentJvm,
+                "Implementation-Title" to archiveBase,
+                "Implementation-Version" to project.version,
+                "Git-Hash" to gitHash,
+                "FMLCorePlugin" to "zmaster587.advancedRocketry.asm.AdvancedRocketryPlugin",
+                "FMLCorePluginContainsFMLMod" to "true"
+        )
+    }
+}
+
+val deobfJar by tasks.registering(Jar::class) {
+    from(sourceSets["main"].output)
+    //if project name is BBM-Dev it makes the deobf jar deobf-dev if not its just deobf
+    if (bambooshortPlanName == "BBM-Dev")
+        archiveClassifier.set("deobf-dev")
+    else
+        archiveClassifier.set("deobf")
+    //sets where the jars go to.
+    destinationDirectory.set(file("output"))
+}
+
+tasks.jar {
+    //if project has the property BBM-Dev then it appends dev to indicate the dev branch
```

Comment:
this whole jar task can be removed. I thought about it and i don't see a need for it anymore.

Reply:
Removed